client/app/scripts/superdesk-search/search.js

```diff
@@ -421,6 +421,23 @@
                         }
                     }
                 });
+                /*
+                * if one of the parameters is the original creator then replace the user id with the
+                * display name.
+                */
+                var i = -1;
+                for (i in tags.selectedParameters) {
+                    if (tags.selectedParameters[i].indexOf('original_creator') >= 0) {
+                        break;
+                    }
+                }
+                if (i >= 0) {
```

Comment:
... actually, if the intent is to find the first matching _selected parameter_, it's probably better to use an existing function than replicating its logic.

```js
var matchingParam = _.find(tags.selectedParameters, function (param) {
    return param.indexOf('original_creator') >= 0;
});

if (matchingParam) {
    var user_id = matchingParam.split(':')[1];
    // etc.
}
```